samples/java-customer-registry/src/main/java/customer/CustomerValueEntityHandler.java

```diff
@@ -16,118 +16,81 @@
-import scala.None$;
-import scala.Option;
-import scalapb.UnknownFieldSet;
-
-import java.lang.reflect.InvocationTargetException;
 
 /**
  * This class would be generated by codegen and should be treated mostly as an implementation detail
  * by the user.
  */
 public class CustomerValueEntityHandler implements ValueEntityHandler {
+
   final CustomerValueEntity entity;
-  final AnySupport anySupport;
 
   public static final Descriptors.ServiceDescriptor serviceDescriptor =
       CustomerApi.getDescriptor().findServiceByName("CustomerService");
   public static final String entityType = "customers";
 
-  CustomerValueEntityHandler(CustomerValueEntity entity) {
-    this.entity = entity;
-    this.anySupport =
-        new AnySupport(
-            new Descriptors.FileDescriptor[] {CustomerDomain.getDescriptor()},
-            CustomerValueEntityHandler.class.getClassLoader(),
-            AnySupport.DefaultTypeUrlPrefix(),
-            AnySupport.PREFER_JAVA());
+  CustomerValueEntityHandler() {
+    this.entity = new CustomerValueEntity();
   }
 
   @Override
-  public ValueEntityBase.Effect<Any> handleCommand(
+  public ValueEntityBase.Effect<? extends GeneratedMessageV3> handleCommand(
       Any command, Any state, CommandContext<Any> context) throws Throwable {
-    AdaptedCommandContext<CustomerDomain.CustomerState> adaptedContext =
-        new AdaptedCommandContext(context, anySupport);
     try {
-      CustomerDomain.CustomerState parsedState =
-          CustomerDomain.CustomerState.parseFrom(state.getValue());
-      // TODO we used to support passing in the command as Jackson-parsed model object
-      // or as ScalaPB class as well. With this change we tie ourselves to Java protobuf.
-      ValueEntityBase.Effect<? extends GeneratedMessageV3> effect =
-          invoke(command, parsedState, adaptedContext);
-      // TODO we used to support accepting ScalaPB and Jackson objects as responses as well.
-      //  Are we OK with losing that? I guess we could have separate methods for that on the
-      //  builder, though it's not obvious how to achieve type-safety for those then.
-
-      // FIXME probably ValueEntityHandler should be adapted to expect an `Effect<? extends
-      // GenerateMessageV3>`?
-      return null;
+      // TODO: we can keep it as CommandContext<Any> for now,
+      //  later we must convert and pass it to the entity before calling the command method
+      return invoke(command, state, context);
     } catch (Exception e) {
-      Throwable unwrapped;
-      if (e.getClass().isAssignableFrom(InvocationTargetException.class) && e.getCause() != null) {
-        unwrapped = e.getCause();
-      } else {
-        unwrapped = e;
-      }
-      if (unwrapped.getClass().isAssignableFrom(FailInvoked$.class)) {
-        throw unwrapped;
+      if (e.getClass().isAssignableFrom(FailInvoked$.class)) {
+        throw e;
       } else {
-        throw unwrapped.getCause();
+        throw e.getCause();
```

Comment:
@raboof, I'm trying to reduce the size of this code. 

My understanding is that InvocationTargetException is a leftover from a reflection based API. We don't need it here anymore. 

So, at the end, either we have a `FailInvoked` that we should throw as is or we have something else in which case we should throw the cause of it. 

Or maybe we should not have any try/catch here and just let whatever exception we get propagate further? WDYT?

Reply:
> My understanding is that InvocationTargetException is a leftover from a reflection based API. We don't need it here anymore

Sounds plausible to me.

> maybe we should not have any try/catch here and just let whatever exception we get propagate further? WDYT?

So you want to move the handling/unwrapping of `FailInvoked` to somewhere else? I have no objection ;)

Reply:
no, my idea is to simple not have the try/catch. We don't unwrap `FailInvoked`, we just rethrow it as is. And for other exceptions we throw the cause, but I wonder if it makes sense. We may just let it bubble up for whatever we get.

Reply:
Aren't we getting rid of FailInvoked from the context.fail anyway, and failing is rather done with `effects().error()`?

Reply:
Of course! :+1:

Reply:
but a user can always throw an exception, no? 

In any case, as suggested in #151, I think we should not have any error handling at all in that part of the code.